### ISSUE TITLE
MapEnvelope does not compare derived classes

### ISSUE THREAD

**Issue by Vatavuk** [yegor256/cactoos#893]
MapEnvelope `equals` method does not support comparing classes derived from `MapEnvelope`.
The following test does not work
```java
@Test
    public void comparesDerivedClass() {
        MatcherAssert.assertThat(
            new TmpMap(),
            new IsEqual<>(new MapOf<>())
        );
    }
    private static class TmpMap extends MapEnvelope<String, String> {

        public TmpMap() {
            super(() -> new HashMap<>());
        }
    }
```
Problem is in the following line:
https://github.com/yegor256/cactoos/blob/62e1fa40c653767850816f3383d27558cdb6e17e/src/main/java/org/cactoos/map/MapEnvelope.java#L151
The following condition could be added:
```java
if (new InheritanceLevel(other.getClass(), MapEnvelope.class).value() < 0) { 
```

**Comment by 0crat** [comment]
@llorllale[/z](https://www.0crat.com/u/llorllale) please, pay attention to this issue

**Comment by llorllale** [comment]
@0crat in

**Comment by 0crat** [comment]
> @0crat in ([here](https://github.com/yegor256/cactoos/issues/893#issuecomment-390672221))

@llorllale Job [#893](https://github.com/yegor256/cactoos/issues/893) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @vatavuk[/z](https://www.0crat.com/u/vatavuk)

**Comment by butenkos** [comment]
@0crat wait for review #893

**Comment by 0crat** [comment]
> @0crat wait for review #925 ([here](https://github.com/yegor256/cactoos/issues/893#issuecomment-404331014))

@butenkos The impediment for [#893](https://github.com/yegor256/cactoos/issues/893) was registered successfully by @butenkos[/z](https://www.0crat.com/u/butenkos)

**Comment by llorllale** [comment]
@0crat resign

**Comment by 0crat** [comment]
> @0crat resign  ([here](https://github.com/yegor256/cactoos/issues/893#issuecomment-443926822))

@llorllale The user @butenkos[/z](https://www.0crat.com/u/butenkos) resigned from [#893](https://github.com/yegor256/cactoos/issues/893), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
The job [#893](https://github.com/yegor256/cactoos/issues/893) assigned to @scratchboom[/z](https://www.0crat.com/u/scratchboom), here is [why](https://www.0crat.com/footprint/C63314D6Z/6dfdee00-c758-4fa0-a9f6-6ae0af19ecc0); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#893` already assigned to @scratchboom, can't assign to @scratchboom

**Comment by llorllale** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/cactoos/issues/893#issuecomment-474141166))

@llorllale The user @scratchboom[/z](https://www.0crat.com/u/scratchboom) resigned from [#893](https://github.com/yegor256/cactoos/issues/893), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
The job [#893](https://github.com/yegor256/cactoos/issues/893) assigned to @borysfan[/z](https://www.0crat.com/u/borysfan), here is [why](https://www.0crat.com/footprint/C63314D6Z/433d4b3b-0736-4fc2-8620-713e53bb154b); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by 0crat** [comment]
The architect of the project has changed; @llorllale[/z](https://www.0crat.com/u/llorllale) is not at this role anymore; @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) is the architect now

**Comment by andreoss** [comment]
@paulodamaso This was fixed in #916

**Comment by victornoel** [comment]
@andreoss thx
